### DISCUSSION TITLE
Overriding enable-admission-plugins in /etc/rancher/rke2/config.yaml results in duplicate lines in apiserver pods.

### DISCUSSION THREAD

**Discussion by easpeagle** [rancher/rke2#3874] [Q&A]
Hey Folks,

I'm trying to reason out why this is happening... if I add the following to `/etc/rancher/rke2/config.yaml`

```
kube-apiserver-arg:
  - enable-admission-plugins=NodeRestriction,PodSecurityPolicy,PodNodeSelector
```
After doing that on all controller nodes and restart rke2-server on each of them... and then restarting the apiserver pods... I see this in those new pods:

```
    - --enable-admission-plugins=NodeRestriction,PodSecurityPolicy
    - --enable-admission-plugins=NodeRestriction,PodSecurityPolicy,PodNodeSelector
```

How do I properly override `enable-admission-plugins` so that it appends `PodNodeSelector` like I want?

**Comment by brandond** [comment]
The user args are appended to the defaults, and Kubernetes takes the last provided value. Your admission plugins configuration should be honored; are you seeing otherwise?

**Reply by easpeagle** [reply]
No... looking at the apicontroller pod logs... they seem to be in there. 

Just odd that rke2 would insert an additional line in the config... rather than overriding or merging defaults...

**Reply by brandond** [reply]
No, we don't attempt to merge or override the args, as Kubernetes itself generally figures it out just fine on its own.

**Reply by easpeagle** [reply]
Well okay then. Confusing for a newb like me then.  Thanks for the clarification.
